File: frontend/types/task.ts
export type TaskPriority = "low" | "medium" | "high";
export type TaskStatus = "todo" | "in_progress" | "done";

export type TaskItem = {
  id: number;
  title: string;
  description: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  due_date: string | null;
  created_by: number;
  assigned_to: number | null;
  department_id: number;
};

export type CreateTaskPayload = {
  title: string;
  description?: string | null;
  priority?: TaskPriority;
  due_date?: string | null;
  department_id: number;
  assigned_to?: number | null;
};

export type UpdateTaskPayload = Partial<CreateTaskPayload>;

export type TaskFormValues = {
  title: string;
  description: string;
  priority: TaskPriority;
  due_date: string;
  department_id: number;
  assigned_to: number | null | "";
};

File: backend/app/schemas/task.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TaskCreate(BaseModel):
    title: str
    description: str | None = None
    status: str = "todo"
    priority: str = "medium"
    due_date: datetime | None = None
    department_id: int
    assigned_to: int | None = None


class TaskUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    status: str | None = None
    priority: str | None = None
    due_date: datetime | None = None
    department_id: int | None = None
    assigned_to: int | None = None


class TaskResponse(BaseModel):
    id: int
    title: str
    description: str | None
    status: str
    priority: str
    due_date: datetime | None
    created_by: int
    assigned_to: int | None
    department_id: int

    model_config = ConfigDict(from_attributes=True)